gammapy/datasets/map.py

```diff
@@ -909,27 +907,64 @@ def to_image(self, spectrum=None, name=None):
             counts = self.counts * mask_safe
             kwargs["counts"] = counts.sum_over_axes(keepdims=True)
 
-        if self.exposure is not None:
-            exposure = _map_spectrum_weight(self.exposure, spectrum)
-            kwargs["exposure"] = exposure.sum_over_axes(keepdims=True)
-
         if self.background_model is not None:
             background = self.background_model.evaluate() * mask_safe
             background = background.sum_over_axes(keepdims=True)
-            kwargs["models"] = Models(
-                [BackgroundModel(background, datasets_names=[name])]
-            )
+        else:
+            background = None
+
+        if isinstance(self.edisp, EDispMap):
+            kwargs["edisp"] = self.edisp
+            kwargs["exposure"] = self.exposure
+            kwargs["psf"] = self.psf
+        else:
+            kwargs["edisp"] = None
+            if self.exposure is not None:
+                exposure = _map_spectrum_weight(self.exposure, spectrum)
+                kwargs["exposure"] = exposure.sum_over_axes(keepdims=True)
 
-        if self.psf is not None:
             # TODO: implement PSFKernel.to_image()
-            if not isinstance(self.psf, PSFKernel):
+            if isinstance(self.psf, PSFMap):
                 kwargs["psf"] = self.psf.to_image(spectrum=spectrum, keepdims=True)
             else:
-                # assume exposure at center position
                 kwargs["psf"] = None
 
+        kwargs["models"] = self._copy_models(background=background, name=name)
+
         return self.__class__(**kwargs)
 
+    def _copy_models(self, background=None, name=None):
+        """
```

Comment:
Here I suppose the problem is more consistency between different `Dataset` transforming methods. 
I probably wouldn't export more than the transformed background model as is the case now. 

We need to define the correct strategy in the long run.